lib/package.gi

```diff
@@ -2282,29 +2375,38 @@ InstallGlobalFunction( ValidatePackageInfo, function( info )
     TestOption( record, "IssueTrackerURL", IsURL, "a string started with http://, https:// or ftp://" );
     TestOption( record, "SupportEmail", IsString, "a string" );
     TestMandat( record, "AbstractHTML", IsString, "a string" );
-    TestMandat( record, "PackageWWWHome", IsURL, "a string started with http://, https:// or ftp://" );
-    if TestMandat( record, "PackageDoc",
-           x -> IsRecord( x ) or IsRecordList( x ),
-           "a record or a list of records" ) then
+    if isSubpackage then
+      TestOption( record, "PackageWWWHome", IsURL, "a string started with http://, https:// or ftp://" );
+    else
+      TestMandat( record, "PackageWWWHome", IsURL, "a string started with http://, https:// or ftp://" );
+    fi;
+    if hasSubpackage then
+      docTester := function( args... ) return IsBound( args[1].(args[2]) ) and CallFuncList( TestOption, args ); end;
+    else
+      docTester := TestMandat;
+    fi;
+    if docTester( record, "PackageDoc",
+          x -> IsRecord( x ) or IsRecordList( x ),
+          "a record or a list of records" ) then
       if IsRecord( record.PackageDoc ) then
         list:= [ record.PackageDoc ];
       else
         list:= record.PackageDoc;
       fi;
       for subrec in list do
-        TestMandat( subrec, "BookName", IsString, "a string" );
+        docTester( subrec, "BookName", IsString, "a string" );
```

Comment:
Why is `docTest` used here and elsewhere? If `record.PackageDoc` is present, then it should pass its full validation check, shouldn't it?

Reply:
Well, because for subpackages, documentation is optional (see comment some lines above). So we use `docTester`, which is either set to `TestMandat` (in case it is not a metapackage) or so something just tests the entries if they are there.

Reply:
That explains the first use of `docTester` in line 2392. But I'd expect a subpackage without manual to have no `PackageDoc` record at all. Conversely, if it has one, shouldn't it be complete?

Reply:
I've now updated the PR to address this, too, as well as a few other minor things. I also squashed most of it in one commit; only the most recent changes I just made are stilling a separate commit, so that you can more easily review them. Assuming you are fine with them, we can squash them into the main commit.